Sources/TrackerRadarKit/ArrayExtension.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  ArrayExtension.swift
+//
+//  Copyright © 2021 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+extension Array where Element: Hashable {
+
+    func removeDuplicates() -> [Element] {
+        var existingElements = Set<Element>()
+        return self.filter { existingElements.insert($0).inserted }
```

Comment:
Similar to the previous code, but now the set is used by the `filter` function to check whether an item should be filtered or not, instead of passing the whole array through the set.